controllers/openstackdataplanenode_controller.go

```diff
@@ -102,10 +102,8 @@ func (r *OpenStackDataPlaneNodeReconciler) Reconcile(ctx context.Context, req ct
 		return ctrl.Result{}, err
 	}
 
-	instanceRole, err := r.GetInstanceRole(ctx, instance)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
+	// An instance can be created without a role so ignore error
+	instanceRole, _ := r.GetInstanceRole(ctx, instance)
```

Comment:
we need to handle the error for when the user provides a wrong name

Reply:
Yes, I agree. So a wrong name might be a role that doesn't exist.

I think I'll rebase this patch on https://github.com/openstack-k8s-operators/dataplane-operator/pull/68 and then handle that case by failing. If the role name isn't set, then by PR 68 the Get won't be run so we need to not fail in that case.

Reply:
if we decide the role is required, I just need to revert that commit

Reply:
I don't think role should be required. Let's make it optional.

Reply:
Agree, it should be optional. Now that we have https://github.com/openstack-k8s-operators/dataplane-operator/pull/68 it effectively is.